app/models/service/aggregation.rb

```diff
@@ -9,12 +9,115 @@ module Service::Aggregation
     virtual_column :aggregate_direct_vm_disk_space_used,      :type => :integer
     virtual_column :aggregate_direct_vm_memory_on_disk,       :type => :integer
 
-    virtual_column :aggregate_all_vm_cpus,                    :type => :integer
-    virtual_column :aggregate_all_vm_memory,                  :type => :integer
-    virtual_column :aggregate_all_vm_disk_count,              :type => :integer
-    virtual_column :aggregate_all_vm_disk_space_allocated,    :type => :integer
-    virtual_column :aggregate_all_vm_disk_space_used,         :type => :integer
-    virtual_column :aggregate_all_vm_memory_on_disk,          :type => :integer
+    virtual_attribute :aggregate_all_vm_cpus,   :integer,
+                      :arel => aggregate_service_hardware_arel('cpus', :cpu_total_cores)
+    virtual_attribute :aggregate_all_vm_memory, :integer,
+                      :arel => aggregate_service_hardware_arel('memory', :memory_mb, :mb_to_bytes => true)
+    virtual_attribute :aggregate_all_vm_disk_count, :integer,
+                      :arel => aggregate_all_vms_disk_count_arel
+    virtual_attribute :aggregate_all_vm_disk_space_allocated, :integer,
+                      :arel => aggregate_all_vms_disk_space_allocated_arel
+    virtual_attribute :aggregate_all_vm_disk_space_used, :integer,
+                      :arel => aggregate_all_vms_disk_space_used_arel
+    virtual_attribute :aggregate_all_vm_memory_on_disk, :integer,
+                      :arel => aggregate_service_hardware_arel('memory_on_disk', :memory_mb, :on_vms_only => true,
+                                                                                             :mb_to_bytes => true)
+
+    AGGREGATE_ALL_VM_ATTRS = [
+      :aggregate_all_vm_cpus,
+      :aggregate_all_vm_memory,
+      :aggregate_all_vm_disk_count,
+      :aggregate_all_vm_disk_space_allocated,
+      :aggregate_all_vm_disk_space_used,
+      :aggregate_all_vm_memory_on_disk
+    ].freeze
+
+    scope :with_aggregates, -> { select(Arel.star, *AGGREGATE_ALL_VM_ATTRS) }
+  end
+
+  module ClassMethods
+    SERVICE_RESOURCES_TBL = ServiceResource.arel_table
+    VMS_TBL               = Vm.arel_table
+    HARDWARES_TBL         = Hardware.arel_table
+    DISKS_TBL             = Disk.arel_table
+    PARTITIONS_TBL        = Partition.arel_table
+
+    VM_OR_TEMPLATE_TYPE   = Arel::Nodes::SqlLiteral.new("'VmOrTemplate'")
+    ANCESTRY_MATCH        = Arel::Nodes::SqlLiteral.new("'/%'")
+    ANCESTRY_ILIKE        = Arel::Nodes::NamedFunction.new("CONCAT", [Service.arel_table[:id], ANCESTRY_MATCH])
```

Comment:
Lets avoid having `arel_table` in constants.
Simply loading this module will force us to hit the db for `Service`

please put these into a lambda or method that is not called at boot time.